playlist_souffle/entities/souffler.py

```diff
@@ -0,0 +1,29 @@
+"""Module for souffler entity."""
```

Comment:
I think I'd call this entity `playlist`, and have it have these public functions:
* `generate_souffled_playlist` (though perhaps just `souffle`)

And these private functions:
* `generate_description`
* `generate_name`

Reply:
So my question is, should an entity called "playlist" be aware that it's a playlist with souffleing logic, or just have business logic related to playlists themselves?

In your suggestion it seems like playlist.generate_name really is playlist.generate_souffle_name. Theres an implicit understanding that anything that takes place in playlist is re: souffleing, which i think is a little confusing (what if playlist.souffle_name is just making a name based on the number of tracks in the playlist or something else?)

In my thought, souffler was the only entity aware of the use case of actually souffleing tracks, etc. Maybe thats wrong though?

Reply:
i'm really not sure. If you were building a music application, burdening playlist with the souffleing seems like too much. but as that's the main thing you're doing, I think having a playlist that's a collection of functions on souffleing playlists makes sense. but ?

Reply:
SOLID principles probably have something to say about this

Reply:
Btw: i've been thinking about this a bit and it seems in the whole clean architecture paradigm, separation of business logic into unique entities seems the least important, especially for something at this scale. (It doesn't impact testability or i/o hoisting at all.)

I'm going to just keep the souffler entity. In my opinion there is a tiny amount of business logic to be done and that can all live in one entity

Reply:
Agreed that the separation between layers is more important the separation within them. And I think keeping the souffler as a single entity is a good call.